docs/latest/design/ratelimit.md

```diff
@@ -0,0 +1,149 @@
+# Rate Limiting Design
+
+## Overview
+
+Rate limiting is a feature that allows the user to limit the number of incoming requests
+to a predefined value based on attributes within the traffic flow.
+
+Here are some reasons why a user may want to implements Rate limits
+
+* To prevent malicious activity such as DDoS attacks.
+* To prevent applications and its resources (such as a database) from getting overloaded.
+* To create API limits based on user entitlements.
+
+## API
+
+* Here is an example of a rate limit implemented by the application developer that limits the total requests made
+to a specific route to safeguard health of internal application components.
+
+```yaml
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: RateLimit
+metadata:
+  name: ratelimit-all-requests
+spec:
+  type: Global
+  rules:
+  - matches:
+    - limit:
+        requests: 1000
+	unit: Second
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example
+spec:
+  parentRefs:
+    - name: eg
+  hostnames:
+    - www.example.com
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /foo
+      filters:
+        - type: ExtensionRef
```

Comment:
will we allow this filter to be applied per-backend? good to mention if that is in or out of scope for this pass

(per-backend ratelimits could be implemented by adding in a descriptor entry to the generated rate-limit for the user)

Reply:
thanks for raising this, this design adds rate limits at a route/traffic flow level, and not per backend (this could be achieved by creating multiple `HTTPRoute`s with a single backend, and attaching a ratelimit filter to each one), will update the design doc with this info